libp2p/protocols/kademlia/dhttypes.nim

```diff
@@ -0,0 +1,46 @@
+import chronicles
+import std/[tables, hashes]
+import results
+
+type EntryKey* = object
+  data*: seq[byte]
+
+type EntryVal* = object
+  data*: seq[byte]
+
+type TimeStamp* = object
+  # Currently a string, because for some reason, that's what is chosen at the protobuf level
+  # TODO: convert between RFC3339 strings and use of integers (i.e. the _correct_ way)
+  ts*: string
+
+type EntryCandidate* = object
+  key*: EntryKey
+  val*: EntryVal
+  time*: TimeStamp
+
+type ValidatedEntry* = object
+  key*: EntryKey
+  val*: EntryVal
+  time*: TimeStamp
+
+type EntryValidator* = ref object of RootObj
+method validate*(
+    self: EntryValidator, key: EntryKey, val: EntryVal
+): bool {.base, raises: [], gcsafe.} =
+  discard
+
+# get a validated entry from a app defined validator and 
+
+type RecordVal* = object
+  value: EntryVal
+  time: TimeStamp
+
+type LocalTable* = object
+  entries*: Table[EntryKey, (EntryVal, TimeStamp)]
+
+proc init*(self: typedesc[LocalTable]): LocalTable {.raises: [].} =
+  LocalTable(entries: initTable[EntryKey, (EntryVal, TimeStamp)]())
```

Comment:
Since we dropped 1.6 support, it is no longer necessary to call initTable from 2.0 onwards according to the docs.

Reply:
nim's stdlib already norms silent default init. calling initTable without a size adds noise (similar to calling newSeq() with no size). Skipping it aligns with idiomatic nim >= 0.20.